tools/autograd/load_derivatives.py

```diff
@@ -85,11 +87,147 @@ def create_derivative(f: NativeFunction, formula: str, var_names: Tuple[str, ...
 
     return Derivative(
         formula=formula,
+        original_formula=original_formula,
         var_names=var_names,
         saved_inputs=saved_inputs,
         saved_outputs=saved_outputs,
     )
 
+def create_forward_derivative(f: NativeFunction, formula: str, names: Tuple[str, ...]) -> ForwardDerivative:
+    assert len(names) == 1, "Forward derivatives can define gradients for only one output at a time"
+    var_name = names[0]
+    var_type: Optional[Type] = None
+    for r in f.func.returns:
+        if r.name == var_name:
+            var_type = r.type
+            break
+    # Handle default return names
+    if var_type is None:
+        if var_name == "result":
+            assert len(f.func.returns) == 1
+            var_type = f.func.returns[0].type
+        else:
+            res = re.findall(r"^result(\d+)$", var_name)
+            if len(res) == 1:
+                arg_idx = int(res[0])
+                var_type = f.func.returns[arg_idx].type
+
+    assert var_type is not None, "No matching output for forward derivative definition"
+    return ForwardDerivative(
+        formula=formula,
+        var_name=var_name,
+        var_type=var_type,
+        required_inputs_fw_grad=None,
+        required_inputs_primal=None)
+
+def postprocess_forward_derivatives(
+    f: NativeFunction,
+    defn_name: str,
+    all_arg_names: List[str],
+    derivatives: List[Derivative],
+    forward_derivatives: List[ForwardDerivative],
+    args_with_derivatives: Sequence[Binding]
+) -> List[ForwardDerivative]:
+
+    def find_required_inputs(formula: str, postfix: str) -> Tuple[str, ...]:
+        required_inputs = set()
+        for arg in args_with_derivatives:
+            if arg.type == 'TensorList':
+                # The functions taking TensorList handle everything internally
```

Comment:
This is my 🧐 face

Reply:
This all seems highly suspect, it seems like you are putting in special cases to the codegen to specifically handle view functions, rather than because what is going on here makes sense with a TensorList.

Reply:
From an engineering perspective, I guess this is OK, but it sure is going to be hella confusing when someone else tries to add a TensorList op to have forward AD and it doesn't work